### PR TITLE
add roxygen quickref

### PR DESCRIPTION
**Pull request by kevinushey**
Adds an `roxygen` + `.Rd` quick-ref guide to the IDE, and adds it to a `?` button for R source files (when within an R package project).

### CODE REVIEW

**Comment by jjallaire** [comment]
Let's discuss realtime later on whether we want to perpetuate the help button on the editor toolbar. I put that in originally because the concept of markdown was so new I wanted a "handlebar" there for users to get oriented. Arguably we could now move both the Markdown Quick Reference and the Roxygen Quick Reference to one or both of the following:

1. The global help menu

2. A new "Ref" button on the Help pane toolbar

Let's discuss when you come in.

**Comment by kevinushey** [comment]
Here's how it looks now:

screen shot 2015-05-19 at 11 18 17 am

With cheat sheet redirects at https://github.com/rstudio/rstudio-web/pull/3.

@jjallaire, I think this is good to go -- any thoughts?

**Comment by rmflight** [comment]
Should the Markdown quick reference stay in the main help? Based on this screenshot it has completely disappeared.

**Comment by rmflight** [comment]
Otherwise this makes sense without adding more clutter to the IDE itself

**Comment by kevinushey** [comment]
The Markdown Quick Reference only displays in the Help screen when the currently open document is a Markdown (or R Markdown) document. This is perhaps a bit confusing?

**Comment by rmflight** [comment]
Yes, yes it is. Didn't realize that context changed in help. I normally expect help to be help, regardless of current file context. On the IDE pane itself this makes sense, different actions are available for different types of files, but I didn't realize that extended to the actual menu as well.

**Comment by jjallaire** [comment]
I agree that we should keep the commands in the help menu all the time. I'd be in favor of the + removing the help button entirely from the C++ toolbar.

I'd also change the caption of the shiny cheat sheet to "Shiny Web Applications"

**Comment by kevinushey** [comment]
@jjallaire, I think this should be good to go now.

File: src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/TextEditingTarget.java [hunk 1]
```diff
@@ -3640,6 +3640,12 @@ void onExecuteLastCode()
    }
    
    @Handler
+   void onKnitDocument()
+   {
+      onPreviewHTML();
+   }
+   
+   @Handler
```
Comment (jjallaire) [inline src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/TextEditingTarget.java]: Doesn't onMarkdownHelp need to be removed from TextEditingTarget?

Comment (kevinushey) [inline src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/TextEditingTarget.java]: You're right, thanks! I missed that.

**Pull request merged**